.storybook/src/complete-example/intersection.stories.ts

```diff
@@ -15,38 +15,70 @@ import {
 
 import { createButtonContainer, createFPSLabel, createLayerContainer, createRootContainer, createHelpText } from '../utils';
 
-import { generateSurfaceData, SurfaceData, getSeismicInfo, generateSeismicSliceImage, transformFormationData, getPicksData } from '../../../src/datautils';
+import {
+  generateSurfaceData,
+  SurfaceData,
+  getSeismicInfo,
+  generateSeismicSliceImage,
+  transformFormationData,
+  getPicksData,
+} from '../../../src/datautils';
 
 //Data
 import { seismicColorMap } from '../exampledata';
 
 import { getCompletion, getSeismic, getSurfaces, getWellborePath, getStratColumns, getHolesize, getCasings, getCement, getPicks } from '../data';
 
-const xBounds: [number, number] = [0, 1000];
-const yBounds: [number, number] = [0, 1000];
+export const intersection = () => {
+  const xBounds: [number, number] = [0, 1000];
+  const yBounds: [number, number] = [0, 1000];
+
+  const scaleOptions = { xBounds, yBounds };
 
-const scaleOptions = { xBounds, yBounds };
-const axisOptions = {
-  xLabel: 'Displacement',
-  yLabel: 'TVD MSL',
-  unitOfMeasure: 'm',
+  return renderIntersection(scaleOptions);
 };
 
-const width = 700;
-const height = 600;
+export const intersectionFlipX = () => {
+  const xBounds: [number, number] = [1000, 0];
+  const yBounds: [number, number] = [0, 1000];
+
+  const scaleOptions = { xBounds, yBounds };
+
+  return renderIntersection(scaleOptions);
+};
+
+const renderIntersection = (scaleOptions) => {
+  const axisOptions = {
+    xLabel: 'Displacement',
+    yLabel: 'TVD MSL',
+    unitOfMeasure: 'm',
+  };
+
+  const width = 700;
+  const height = 600;
 
-export const intersection = () => {
   // helper container elements
   const root = createRootContainer(width);
   const btnToggleContainer = createButtonContainer(width);
   const btnAdjustSizeContainer = createButtonContainer(width);
   const btnMiscContainer = createButtonContainer(width);
   const container = createLayerContainer(width, height);
 
-  const promises = [getWellborePath(), getCompletion(), getSeismic(), getSurfaces(), getStratColumns(), getCasings(), getHolesize(), getCement(), getPicks()];
+  const promises = [
+    getWellborePath(),
+    getCompletion(),
+    getSeismic(),
+    getSurfaces(),
+    getStratColumns(),
+    getCasings(),
+    getHolesize(),
+    getCement(),
+    getPicks(),
+  ];
```

Comment:
What if any of the elements are rejected? Nothing is done right now. Should that be logged or something?

Reply:
Since all comes from same source I guess they will all fail or come through. Since this is just in the storybook I'm not sure we need to build in logic for that.